### PR TITLE
workflows: add redhat and community workflow logic

### PR DESCRIPTION
**Pull request by zonggen**
As part of supporting different vendor types for openshift helm charts, this adds the github actions for redhat and community
submission. For design details please refer to: docs/workflow-build.md.

Signed-off-by: Allen Bai <abai@redhat.com>

### CODE REVIEW

**Comment by zonggen** [comment]
/hold

Working on updating the documentation.

File: scripts/src/chartprreview/chartprreview.py
Comment (dperaza4dustbit) [inline scripts/src/chartprreview/chartprreview.py:260]: nice, good to know

File: scripts/src/chartprreview/chartprreview.py
Comment (dperaza4dustbit) [inline scripts/src/chartprreview/chartprreview.py:269]: maybe keep all the messages outside this if? It is good data to have even in the event that we force-publish

File: scripts/src/chartprreview/chartprreview.py
Comment (dperaza4dustbit) [inline scripts/src/chartprreview/chartprreview.py:305]: Maybe another message since for the user perspective this means they did not place the chart or report in the right directory.

File: scripts/src/chartprreview/verify-report.sh
Comment (dperaza4dustbit) [inline scripts/src/chartprreview/verify-report.sh:25]: Since the same right just do mandatoryChecksRedHat=mandatoryChecksPartner

File: scripts/src/chartrepomanager/chartrepomanager.py
Comment (dperaza4dustbit) [inline scripts/src/chartrepomanager/chartrepomanager.py:148]: good

**Review by dperaza4dustbit — COMMENTED** [review]
This loos like solid logic, just a few suggestions.

File: scripts/src/chartprreview/chartprreview.py
Comment (zonggen) [reply scripts/src/chartprreview/chartprreview.py:269]: yea better not to silence the error, will update

File: scripts/src/chartprreview/chartprreview.py
Comment (zonggen) [reply scripts/src/chartprreview/chartprreview.py]: updated

**Review by dperaza4dustbit — COMMENTED** [review]
Some minor comments. I also thing we need to update the docs/README.md file to make sure redhat and community chart owners know how to send PR first for OWNERs file and then place the chart in the right directory. Let me know if you need help with the text.

File: .github/workflows/build.yml
Comment (zonggen) [reply .github/workflows/build.yml:122]: For reference, as per the sync call, we are planning to use `main` for now and make update once we have the new `latest` release.

File: scripts/src/chartprreview/chartprreview.py
Comment (zonggen) [reply scripts/src/chartprreview/chartprreview.py:305]: Here the vendor type is used by the verifier docker run command. Will add a helper getter to reduce repeated code. Nice suggestion!

File: docs/workflow-build.md
Comment (dperaza4dustbit) [reply docs/workflow-build.md:10]: yeah jus stop here ".... update the PR according to the report to pass the checks."

File: scripts/src/chartprreview/chartprreview.py
Comment (zonggen) [reply scripts/src/chartprreview/chartprreview.py]: updated

File: docs/workflow-build.md
Comment (zonggen) [reply docs/workflow-build.md:10]: updated

File: scripts/src/chartprreview/chartprreview.py
Comment (zonggen) [reply scripts/src/chartprreview/chartprreview.py]: Sorry but I would rather have an `if` just to be explicit about the condition..

File: scripts/src/chartprreview/chartprreview.py
Comment (zonggen) [reply scripts/src/chartprreview/chartprreview.py]: updated

File: .github/workflows/build.yml
Comment (dperaza4dustbit) [reply .github/workflows/build.yml:122]: agree

**Review by dperaza4dustbit — APPROVED** [review]
lgtm, OK to merge to dev. More testing needed there to cherrypick to main

File: scripts/src/chartprreview/chartprreview.py
Comment (dperaza4dustbit) [reply scripts/src/chartprreview/chartprreview.py]: OK, your call

File: docs/workflow-build.md
Comment (zonggen) [reply docs/workflow-build.md]: updated

File: docs/workflow-build.md
Comment (zonggen) [reply docs/workflow-build.md]: updated

**Review by mmulholla — APPROVED** [review]

**Pull request merged**
